### PR TITLE
feat!: resource generation logic fix

### LINKED ISSUE #224: dlp - unknown resource
I noticed that the DLP baseline tests print this:

```
  LibraryWithMultiplePatternResourceGenerateTest
    Generate Client library
Warning: service DlpService message .google.privacy.dlp.v2.CreateInspectTemplateRequest refers to an unknown resource: dlp.googleapis.com/InspectTemplate
Warning: service DlpService message .google.privacy.dlp.v2.ListInspectTemplatesRequest refers to an unknown resource: dlp.googleapis.com/InspectTemplate
Warning: service DlpService message .google.privacy.dlp.v2.CreateDeidentifyTemplateRequest refers to an unknown resource: dlp.googleapis.com/DeidentifyTemplate
Warning: service DlpService message .google.privacy.dlp.v2.ListDeidentifyTemplatesRequest refers to an unknown resource: dlp.googleapis.com/DeidentifyTemplate
Warning: service DlpService message .google.privacy.dlp.v2.CreateStoredInfoTypeRequest refers to an unknown resource: dlp.googleapis.com/StoredInfoType
Warning: service DlpService message .google.privacy.dlp.v2.ListStoredInfoTypesRequest refers to an unknown resource: dlp.googleapis.com/StoredInfoType
      ✓ Generated client library with multiple pattern resource should have same output with baseline. (668ms)
```

It's a little bit unclear since all mentioned resources are defined in `dlp.proto`. Something going on with multiple patterns? @xiaozhenliu-gg5 can you please take a look?

### PR DESCRIPTION
**Pull request by xiaozhenliu-gg5**
some common resource path are missing in the pathTemplate and improper warnings.

fix #224

### CODE REVIEW

File: typescript/src/schema/api.ts [hunk 1]
```diff
@@ -159,7 +159,11 @@ function getResourceDatabase(
...
         `file ${fd.name} message ${property}`
       );
+      allResourceDatabase.registerResource(
+        m?.options?.['.google.api.resource'] as ResourceDescriptor | undefined,
+        `file ${fd.name} message ${property}`
+      );
     }
   }
-  return [resourceDatabase, resourceDefinitionDatabase];
+  return [allResourceDatabase, resourceDatabase];
```
Comment (alexander-fenster) [inline typescript/src/schema/api.ts]: We need to fix these names :)

Comment (xiaozhenliu-gg5) [reply typescript/src/schema/api.ts]: Sure, was hesitating about a better name, thanks for the suggestion!

**Comment by xiaozhenliu-gg5** [comment]
Basically now we have two databases: one for resources defined by `google.api.resource` and the other one for all resources, both `google.api.resource` and `google.api.resource_definition`. So I may have to come up with a name the differentiate `google.api.resource`.

**Comment by xiaozhenliu-gg5** [comment]
should unlock: https://github.com/googleapis/nodejs-security-center/pull/198

**Pull request merged**